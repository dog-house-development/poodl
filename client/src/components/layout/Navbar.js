import React, { Component } from "react";
import { Link } from "react-router-dom";

class Navbar extends Component {
  render() {
    return (
      <div className="navbar-fixed">
        <nav className="z-depth-0">
          <div className="nav-wrapper white">
            <Link
              to="/" >
              Poodl
            </Link>
          </div>
        </nav>
      </div>
    );
  }
}

export default Navbar;
