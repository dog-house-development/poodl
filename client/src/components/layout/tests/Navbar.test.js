import React from "react";
import ReactDOM from "react-dom";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Navbar from "../Navbar";

configure({ adapter: new Adapter() });

describe("Navbar", () => {
  let props, wrapper, instance;
  beforeEach(() => {
    props = {
      content: "Press me",
      onClick: () => {}
    };
    wrapper = shallow(<Navbar {...props} />);
    instance = wrapper.instance();
  });

  it("should render correctly", () => {
    expect(instance).toMatchSnapshot();
  });
});
