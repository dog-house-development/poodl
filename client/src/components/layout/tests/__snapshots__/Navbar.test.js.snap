// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests render should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    content="Press me"
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="z-depth-0"
      >
        <div
          className="nav-wrapper white"
        >
          <Link
            replace={false}
            to="/"
          >
            Poodl
          </Link>
        </div>
      </nav>,
      "className": "navbar-fixed",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nav-wrapper white"
        >
          <Link
            replace={false}
            to="/"
          >
            Poodl
          </Link>
        </div>,
        "className": "z-depth-0",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/"
          >
            Poodl
          </Link>,
          "className": "nav-wrapper white",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Poodl",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Poodl",
          "type": [Function],
        },
        "type": "div",
      },
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="z-depth-0"
        >
          <div
            className="nav-wrapper white"
          >
            <Link
              replace={false}
              to="/"
            >
              Poodl
            </Link>
          </div>
        </nav>,
        "className": "navbar-fixed",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="nav-wrapper white"
          >
            <Link
              replace={false}
              to="/"
            >
              Poodl
            </Link>
          </div>,
          "className": "z-depth-0",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
            >
              Poodl
            </Link>,
            "className": "nav-wrapper white",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Poodl",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Poodl",
            "type": [Function],
          },
          "type": "div",
        },
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
